R/utils.R

```diff
@@ -327,4 +335,66 @@ valid_seq <- function(params) {
   } else {
     return(FALSE)
   }
-}
\ No newline at end of file
+}
+
+resolve_prefix <- function(prefix, environment_var) {
+  if (!(is.null(prefix))) {
+    assertthat::assert_that(is.character(prefix))
+    
+    return(prefix)
+  } else {
+    prefix_env <- Sys.getenv(environment_var)
+    if (prefix_env != "") {
+      return(prefix_env)
+    } else {
+      return("/")
+    }
+  }
+}
+
+get_package_mapping <- function(script_path, url_package, dependencies) {
+  package_map <- vapply(dependencies, function(x) {
+    if (x$name %in% c('react', 'react-dom')) {
+      x$name <- 'dash_renderer'
+    }
+    dep_path <- paste(x$src$file, 
+                      x$script,
+                      sep = "/")
+  
+    # remove double slash and replace with single if present
+    result <- c(pkg_name=x$package,
+                dep_name=x$name,
+                dep_path=gsub("//", replacement = "/", dep_path)
+    )
+  }, FUN.VALUE = character(3))
+  
+  package_map <- as.data.frame(t(package_map), stringsAsFactors = FALSE)
```

Comment:
could we add a comment here? (why converting to a dataframe?)

Reply:
I suspect this is a holdover from an earlier version of the function in which strings were being coerced to factors and transposing the table was leading to some strange behaviour. I think I can safely replace this with

```suggestion
package_map <- t(package_map)
```

Reply:
fixed in b34c4f841fc0fe45f9c527823a98de97c3778394